great_expectations/datasource/pandas_datasource.py

```diff
@@ -213,14 +214,10 @@ def get_batch(self, batch_kwargs, batch_parameters=None):
             )
             s3_object = s3.get_object(Bucket=url.bucket, Key=url.key)
             reader_fn = self._get_reader_fn(reader_method, url.key)
-            df = reader_fn(
-                StringIO(
-                    s3_object["Body"]
-                    .read()
-                    .decode(s3_object.get("ContentEncoding", "utf-8"))
-                ),
-                **reader_options
-            )
+            if not reader_options.get("encoding"):
+                reader_options["encoding"] = s3_object.get("ContentEncoding", "utf-8")
+            reader_options = self._infer_default_options(reader_fn, reader_options)
```

Comment:
As written, this will override reader_options explicitly passed by the user, I believe.

```suggestion
            default_reader_options = self._infer_default_options(reader_method, reader_options)
            if not reader_options.get("encoding") and default_reader_options.get("encoding"):
                reader_options["encoding"] = s3_object.get("ContentEncoding", default_reader_options.get("encoding"))
```

Reply:
Good catch here. My mistake. I'll update!